workspaces/src/rpc/client.rs

```diff
@@ -229,22 +236,27 @@ impl Client {
                 block_reference,
                 request: QueryRequest::ViewCode { account_id },
             })
-            .await?;
+            .await
+            .map_err(|e| RpcErrorKind::QueryFailure.with_repr(e.into()))?;
 
         match query_resp.kind {
             QueryResponseKind::ViewCode(code) => Ok(code),
-            _ => anyhow::bail!(ERR_INVALID_VARIANT),
+            _ => Err(RpcErrorKind::QueryReturnedInvalidData.with_msg("while querying code")),
         }
     }
 
-    pub(crate) async fn view_block(&self, block_id: Option<BlockId>) -> anyhow::Result<BlockView> {
+    pub(crate) async fn view_block(
+        &self,
+        block_id: Option<BlockId>,
+    ) -> Result<BlockView, RpcError> {
         let block_reference = block_id
             .map(Into::into)
             .unwrap_or_else(|| Finality::None.into());
 
         let block_view = self
             .query(&methods::block::RpcBlockRequest { block_reference })
-            .await?;
+            .await
+            .map_err(|e| RpcErrorKind::QueryBlockFailure.with_repr(e.into()))?;
```

Comment:
Do you have some kind of criteria why some of the calls have their own Failure type (e.g. `QueryBlockFailure`), but others use a generic one (`QueryFailure`)?

Reply:
ahh, that's a good point! They don't really make much of a difference. They can all just be `QueryFailure`s, unless you think they should all be more explicit like `QueryBlockFailure`? Changed to `QueryFailure` in latest commit